libs/filesystem/include/hpx/filesystem.hpp

```diff
@@ -0,0 +1,76 @@
+//  Copyright (c) 2019 Mikael Simberg
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+/// \file
+/// This file provides a compatibility layer using Boost.Filesystem for the
+/// C++17 filesystem library. It is *not* intended to be a complete
+/// compatibility layer. It only contains functions required by the HPX
+/// codebase. It also provides some functions only available in Boost.Filesystem
+/// when using C++17 filesystem.
+
+#if !defined(HPX_FILESYSTEM_HPP)
+#define HPX_FILESYSTEM_HPP
+
+#include <hpx/config.hpp>
+
+#if defined(HPX_HAVE_CXX17_FILESYSTEM)
+#include <filesystem>
+#include <system_error>
+
+namespace hpx { namespace filesystem {
+    using namespace std::filesystem;
+    using std::error_code;
+    using std::filesystem::canonical;
+
+    inline path initial_path()
+    {
+        static path ip = current_path();
+        return ip;
+    }
+
+    inline path basename(path p)
+    {
+        return p.stem();
+    }
+
+    inline path canonical(path p, path base)
```

Comment:
Couldn't these be passed by `const&` instead?

Reply:
Yep, and should probably be passed like that as well. That's what the `std::filesystem::canonical` takes anyway. Good catch, thanks!